### PR TITLE
Fix expectations in kids

### LINKED ISSUE #1046: Expectation results are never shown in kids layout
You are just told that "you can do it better", but there is not feedback about it, nor you can not try again

image

### PR DESCRIPTION
**Pull request by fedescarpa**
Closes #1046, we are showing only the first failed expectations

image

### CODE REVIEW

File: app/assets/javascripts/application/kids.js [hunk 1]
```diff
@@ -171,6 +172,10 @@ mumuki.load(function () {
       $bubble.find('.mu-kids-character-speech-bubble-normal').hide();
       $bubble.find('.mu-kids-character-speech-bubble-failed').show().html(data.title_html);
       $bubble.addClass(data.status);
+      console.log(data);
```
Comment (luchotc) [inline app/assets/javascripts/application/kids.js]: :open_mouth:

File: app/views/layouts/_kids.html.erb [hunk 2]
```diff
@@ -14,5 +14,5 @@
     <div class="description"><%= @exercise.description_task %></div>
     <div class="hint" style="display: none"><%= @exercise.hint_html %></div>
   </div>
-  <div class="mu-kids-character-speech-bubble-failed" style="display: none"></div>
+  <div class="mu-kids-character-speech-bubble-failed <%= @assignment.status if @assignment %>" style="display: none"></div>
```
Comment (luchotc) [inline app/views/layouts/_kids.html.erb]: You could use @assignment&.status

**Comment by flbulgarelli** [comment]
Build is broken because `mumuki-styles` changed paths in an incompatible way. See https://github.com/mumuki/mumuki-laboratory/pull/1097#issuecomment-398766058

**Comment by fedescarpa** [comment]
@flbulgarelli, I resolve the problem rebasing the branch to master with mumki-styles changes

File: app/views/exercise_solutions/_expectations.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<% expectation = assignment.visible_expectation_results.find { |it| it[:status] != :passed } %>
```
Comment (flbulgarelli) [inline app/views/exercise_solutions/_expectations.html.erb]: use `passed?` message: `!it[:status].passed?`. This abstract you from dealing with strings or symbols.

Comment (flbulgarelli) [reply app/views/exercise_solutions/_expectations.html.erb]: PS: sorry for not detecting this before :pensive:

Comment (fedescarpa) [reply app/views/exercise_solutions/_expectations.html.erb]: `passed?` is true when status is passed or passed_with_warnings.

I need to be true only when is passed and false if passed_with_warnings

Comment (flbulgarelli) [reply app/views/exercise_solutions/_expectations.html.erb]: Good point!

But I think this is only true for `Status` object -  which is wrong and inconsistent, I think - but not for strings or symbols - and I understand this is the case here, right?

```ruby
[1] pry(main)> :passed.passed?
=> true
[2] pry(main)> :passed_with_warnings.passed?
=> false
[3] pry(main)> 
```

**Comment by fedescarpa** [comment]
@flbulgarelli, done!

**Pull request merged**